src/app/shared/pipes/search-by/search-by.pipe.ts

```diff
@@ -0,0 +1,21 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+export interface SearchByPipeParams {
+  field: string,
+  searchString: string,
+}
+
+@Pipe({
+  name: 'searchBy',
+})
+export class SearchByPipe implements PipeTransform {
+  transform(array: any[], { field, searchString }: SearchByPipeParams): any[] {
+    if (!Array.isArray(array)) {
+      return [];
+    }
+
+    const filteredArray = array.filter(item => item[field].toLowerCase().includes(searchString.toLowerCase()));
```

Comment:
what will be in case when field or search string is not provided

Reply:
@boale, usage of interface with mandatory fields forces developer to pass all necessary parameters
![image](https://user-images.githubusercontent.com/34865063/99300320-1b488200-2855-11eb-9fe5-7370817fcf6a.png)

Reply:
@boale, now I got you, the interface only protects when this pipe is called from component class. In case of calling pipe from markup this interface doesn't make sense. I'll add default `field`, `searchString` values

Reply:
yep, but it is dees not help when code is compiled and deployed  to environment...

Reply:
adding such a small check will prevent your app from falling in case when you get corrupted data.